ddl/ddl.go

```diff
@@ -1227,12 +1634,32 @@ func GetHistoryJobByID(sess sessionctx.Context, id int64) (*model.Job, error) {
 	return job, errors.Trace(err)
 }
 
-// AddHistoryDDLJob adds DDL job to history table.
-func AddHistoryDDLJob(t *meta.Meta, job *model.Job, updateRawArgs bool) error {
+// AddHistoryDDLJobForTest used for test.
+func AddHistoryDDLJobForTest(sess sessionctx.Context, t *meta.Meta, job *model.Job, updateRawArgs bool) error {
+	return AddHistoryDDLJob(newSession(sess), t, job, updateRawArgs, variable.EnableConcurrentDDL.Load())
+}
+
+// AddHistoryDDLJob record the history job.
+func AddHistoryDDLJob(sess *session, t *meta.Meta, job *model.Job, updateRawArgs bool, _ bool) error {
```

Comment:
Do we miss check`variable.EnableConcurrentDDL.Load()` here?

Reply:
no, adding history will always add history job into the table and history key

Reply:
OK, I got it. Maybe we can add a comment here, on why we do this.

Reply:
comments updated